atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -329,58 +330,49 @@ def resolve_signing_intent(self):
         self.compose_config.set_signing_intent(signing_intent['name'])
 
     def forward_composes(self):
-        repos_by_arch = defaultdict(list)
-        # set overrides by arch if arches are available
         for compose_info in self.composes_info:
             result_repofile = compose_info['result_repofile']
             try:
                 arches = compose_info['arches']
             except KeyError:
-                repos_by_arch[None].append(result_repofile)
+                self.yum_repourls['noarch'].add(result_repofile)
             else:
                 for arch in arches.split():
-                    repos_by_arch[arch].append(result_repofile)
+                    self.yum_repourls[arch].add(result_repofile)
 
         # we should almost never have a None entry from composes,
         # but we can have yum_repos added, so if we do, we need to merge
         # it with all other repos.
         if self.repourls:
-            repos_by_arch[None].extend(self.repourls)
-        try:
-            noarch_repos = repos_by_arch.pop(None)
-        except KeyError:
-            pass
-        else:
-            for repos in repos_by_arch.values():
-                repos.extend(noarch_repos)
-
-        for arch, repofiles in repos_by_arch.items():
-            # OSBS2 TBD: `override_build_kwarg` is imported from build_orchestrate_build
-            override_build_kwarg(self.workflow, 'yum_repourls', repofiles, arch)
-
-        # Only set the None override if there are no other repos
-        if not repos_by_arch:
-            # OSBS2 TBD: `override_build_kwarg` is imported from build_orchestrate_build
-            override_build_kwarg(self.workflow, 'yum_repourls', noarch_repos, None)
+            for repourl in self.repourls:
+                self.yum_repourls['noarch'].add(repourl)
+        if 'noarch' in self.yum_repourls:
+            noarch_repos = self.yum_repourls.pop('noarch')
+            for arch in self.yum_repourls:
+                for repo in noarch_repos:
+                    self.yum_repourls[arch].add(repo)
```

Comment:
again use union

Reply:
updated to use `.update()` since union wasn't working